### PR TITLE
Add conditional loading of streamwrapper

### PR DESCRIPTION
**Pull request by hanifn**
## Description

Load either the stream wrapper class and plugin filters or a8c-files based on a predefined constant.

The constant, `VIP_FILESYSTEM_USE_STREAM_WRAPPER`, should be defined in somewhere like `wp-config.php` where it's loaded before the MU plugins.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.

## Steps to Test

Test that media upload functions work with and without the constant being defined.

### CODE REVIEW

File: a8c-files.php [hunk 1]
```diff
@@ -902,23 +905,30 @@ function a8c_files_maybe_inject_image_sizes( $data, $attachment_id ) {
 }
 
 if ( defined( 'FILES_CLIENT_SITE_ID' ) && defined( 'FILES_ACCESS_TOKEN' ) ) {
-	add_action( 'init', 'a8c_files_init' );
-	add_filter( 'intermediate_image_sizes', 'wpcom_intermediate_sizes' );
-	add_filter( 'intermediate_image_sizes_advanced', 'wpcom_intermediate_sizes' );
-	add_filter( 'fallback_intermediate_image_sizes', 'wpcom_intermediate_sizes' );
+	// Conditionally load either the new Stream Wrapper implementation or old school a8c-files
+	if ( defined( 'VIP_FILESYSTEM_USE_STREAM_WRAPPER' ) ) {
```
Comment (mjangda) [inline a8c-files.php]: We need to move this inside `a8c_files_init` instead. The code for `intermediate_image_sizes` and `srcset` below are relevant in both situations.

Comment (mjangda) [reply a8c-files.php]: Should be fixed with 3537180 but needs more local and production testing.

**Comment by mjangda** [comment]
Some things we'll need to restore for stream wrappers in future PRs:

- File type checks on upload (`wp_check_filetype_and_ext` + `A8C_Files::filter_filetype_check`)
- Purge on delete (`wp_delete_file` + `A8C_Files::purge_file_cache`)

**Comment by hanifn** [comment]
Thanks Mo!

**Comment by mjangda** [comment]
r125274-deploy

**Pull request merged**